File: enalotto.pl
#!/usr/bin/perl

use 5.16.0;
use warnings;

my %data = ();
my @keys = keys %data;

while(@keys < 6) {
	$data{1 + int(rand(90))} = 1;
	@keys = keys %data;
}

my @numbers = sort { $a <=> $b } @keys;

while (my ($index, $number) = each @numbers) {
	print "0" if ($number < 10);
	print "$number";
	print " - " if ($index < 5);
}

print "\n";

File: enalotto.rb

require 'set'

numbers = SortedSet.new

while numbers.size() < 6  do
	numbers.add(rand(90) + 1)
end

numbers.each_with_index do |number, index|
	print "0" if number < 10
	print number
	print " - " if index < 5
end

puts ""